tests/Gedmo/Mapping/MappingEventAdapterTest.php

```diff
@@ -12,7 +12,7 @@
 namespace Gedmo\Tests\Mapping;
 
 use Doctrine\ORM\EntityManager;
-use Doctrine\ORM\Event\LifecycleEventArgs;
+use Doctrine\Persistence\Event\LifecycleEventArgs;
```

Comment:
There is this code 😞 

https://github.com/doctrine-extensions/DoctrineExtensions/blob/351caed07e47552ab0df63cafb4fdcccc009eb59/src/Mapping/MappedEventSubscriber.php#L252-L277

that do some funny things to determine which event adapter has to be instantiated and it's based on the event class namespace that should include `ORM` or `ODM`.

It would be awesome to be able to get rid of this code, but since it doesn't seem easy, maybe we should keep using the deprecated `Doctrine\Persistence\Event\LifecycleEventArgs;`, once https://github.com/doctrine-extensions/DoctrineExtensions/pull/2669 is merged we can use one of the new dedicated events.

Reply:
GitHub doesn't have the correct emoji to express my feelings about that code. Sometimes I feel like code reflection goes just a little too far in some languages, hahaha.

Anyway, I've reverted back to Doctrine\ORM\Event\LifecycleEventArgs.

Reply:
Regarding `getEventAdapter()`, I stumbled with that yesterday in my tests :disappointed_relieved: 

>GitHub doesn't have the correct emoji to express my feelings about that code. Sometimes I feel like code reflection goes just a little too far in some languages, hahaha.

:dizzy_face: :face_with_spiral_eyes: :rofl: 

>It would be awesome to be able to get rid of this code, but since it doesn't seem easy, maybe we should keep using the deprecated Doctrine\Persistence\Event\LifecycleEventArgs;, once https://github.com/doctrine-extensions/DoctrineExtensions/pull/2669 is merged we can use one of the new dedicated events.

Just for the records, `Doctrine\ORM\Event\LifecycleEventArgs` is deprecated, but `Doctrine\Persistence\Event\LifecycleEventArgs` is not.